cfgov/cfgov/settings/production.py

```diff
@@ -114,3 +115,10 @@
         'TIMEOUT': None,
     }
 }
+
+# ALLOWED_HOSTS should be defined as a JSON list in the ALLOWED_HOSTS
+# environment variable.
+try:
+    ALLOWED_HOSTS = json.loads(os.getenv('ALLOWED_HOSTS'))
+except (TypeError, ValueError):
```

Comment:
Looking at this again, I'm wondering about this fallback case. In true "production" deployments, we need to make sure that our configuration (Ansible) sets this environment variable properly. If something goes wrong there and it's not set, then falling back to this wildcard actually seems wrong, as it would both allow hosts we don't want (like beta on www) and prevent hosts we do (like localhost for same-server requests).

Similarly, there are infrequent times when it's useful as a developer to run against these production settings, and the typical `runserver` workflow wouldn't include a valid `localhost` host header in that case.

If we're going to rely on this environment variable, does it make sense to require it to be set, and add it to `.env_SAMPLE` with some reasonable default value (`localhost`)? Then we could add a more verbose error message, something like (just spitballing):

```py
# settings.py
from django.core.exceptions import ImproperlyConfigured

try:
    ALLOWED_HOSTS = json.loads(os.getenv('ALLOWED_HOSTS'))
except (TypeError, ValueError):
    raise ImproperlyConfigured(
        "Use of production settings requires a valid ALLOWED_HOSTS environment "
        "variable containing a JSON list of allowed hostnames."
    )

Reply:
I was purposefully leaving it out of `base.py` and `.env_SAMPLE` (and the `.python_env_SAMPLE` for docker)  because locally we use `DEBUG = True` which ignores `ALLOWED_HOSTS` generally. So in order to test this setting locally you have to modify the settings to change `DEBUG` anyway.

Replacing the default with an `ImproperlyConfigured` exception is a good idea, I think the rationale for doing so makes sense.

Reply:
Done!